dynamic_programming/viterbi.nim

```diff
@@ -1,4 +1,14 @@
 # Viterbi
+## The Viterbi algorithm is a dynamic programming algorithm
+## for obtaining the maximum a posteriori probability estimate
+## of the most likely sequence of hidden states
+## https://en.wikipedia.org/wiki/Viterbi_algorithm
+##
+## Time Complexity: O(K2n)
+## Space Complexity: O(Kn)
```

Comment:
What is K and what is n?
Why one is lowercase and not the other?
Drop the constant in the time complexity.